c++/src/kj/async.c++

```diff
@@ -878,6 +886,11 @@ struct FiberStack::StartRoutine {
     ptr |= static_cast<uintptr_t>(static_cast<uint>(arg2)) << (sizeof(ptr) * 4);
     auto stack = reinterpret_cast<FiberStack*>(ptr);
 
+    /// Jump back into the fiber stack constructor
+    if (setjmp(&stack->impl.fiberContext.jumpBuf[0]) == 0) {
+      swapcontext(&stack->impl.fiberContext.context, &stack->impl.originalContext.context);
```

Comment:
This can be `setcontext()` rather than `swapcontext()` as we don't use `fiberContext.context` anymore after this point.

Though better yet, I think you could be using `longjmp()` here too to return to the calling fiber, if `FiberStack`'s constructor were to call `setjmp()`. And then you could use `setcontext()` rather than `swapcontext()` there.

Then you can go another step further: You only ever really need a `ucontext_t` when you are calling `makecontext()` to create the fiber initially. You could allocate that `ucontext_t` on the stack inside `FiberStack`'s constructor, rather than put it into `FastContext`. Then you could remove `FastContext::context` entirely; you only need `jumpBuf` there.

Let me know if that makes sense...

Reply:
Set context stack info, make the context, setjmp, set the context to start, longjmp back. Got it.